<launch>
  <arg name="input_cloud" default="/kinect2/depth_highres/points"/>
  <arg name="input_image" default="/kinect2/rgb/image"/>

  <arg name="custom" default="true"/>
  <arg name="manager" default="passthrough_manager"/>
  <arg name="rect" default="false" />

  <node pkg="pr2_navigation_self_filter"
        type="self_filter" clear_params="true"
        name="robot_shadow_self_filter2" respawn="true" output="screen">
    <!-- The topic for the input cloud -->
    <remap from="cloud_in" to="$(arg input_cloud)" />
    <!-- The topic for the output cloud -->
    <remap from="cloud_out" to="$(arg input_cloud)_filtered" />
    <param name="subsample_value" type="double" value="0.0"/>
    <param name="use_rgb" type="bool" value="true"/>
    <param name="keep_organized" type="bool" value="true"/>
    <!-- The names of the links the sensor can see -->
    <rosparam if="$(arg custom)" command="load" file="$(find jsk_2014_picking_challenge)/config/openni_self_filter_custom.yaml" />
    <rosparam unless="$(arg custom)" command="load" file="$(find jsk_baxter_startup)/jsk_baxter_sensors/openni_self_filter.yaml" />
  </node>

  <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" />


  <node pkg="nodelet" type="nodelet" name="left_right_divider"
        args="load jsk_pcl/AttentionClipper $(arg manager)" respawn="true">
    <remap from="~input/points" to="/kinect2/depth_highres/points_filtered" />
    <!-- suppose QVGA -->
    <rosparam>
      use_multiple_attention: true
      initial_pos_list: [[0,0,0.3], [0,0,0.3]]
      initial_rot_list: [[0,0,0], [0,0,0]]
      dimensions: [[0.5, 0.5, 0.5],[0.5, 0.5, 0.5]]
      frame_id_list: [right_hand, left_hand]
      prefixes: [right_hand, left_hand]
    </rosparam>
  </node>

  <group ns="left_process">
    <include file="$(find jsk_2014_picking_challenge)/launch/process_mask_image.launch">
      <arg name="manager" value="$(arg manager)"/>
      <arg name="original_image" value="$(arg input_image)"/>
      <arg name="input_point_indices" value="/left_right_divider/left_hand/point_indices"/>
      <arg name="input_depth_camera_info" value="/kinect2/depth_highres/camera_info"/>
      <arg name="rect" value="$(arg rect)"/>
    </include>
  </group>

  <group ns="right_process">
    <include file="$(find jsk_2014_picking_challenge)/launch/process_mask_image.launch">
      <arg name="manager" value="$(arg manager)"/>
      <arg name="original_image" value="$(arg input_image)"/>
      <arg name="input_point_indices" value="/left_right_divider/right_hand/point_indices"/>
      <arg name="input_depth_camera_info" value="/kinect2/depth_highres/camera_info"/>
      <arg name="rect" value="$(arg rect)"/>
    </include>
  </group>



</launch>
